src/resources.rs

```diff
@@ -49,6 +80,7 @@ impl Resources {
     }
 }
 
+/// An immutable reference to a resource.
```

Comment:
Personally, I would prefer "borrow" instead of "reference" for this and the analog `Comp` to emphasise that this type facilitates runtime borrow checking.

Reply:
Replaced at all occurences